src/lib.rs

```diff
@@ -1576,6 +2168,195 @@ mod test {
             test_assign(_1_2, 1, _3_2);
         }
 
+        #[test]
+        fn test_primitive_add_rational() {
+            fn test_isize(a: isize, b: Rational, c: Rational) {
+                assert_eq!(a + b, c);
+            }
+            fn test_i8(a: i8, b: Rational, c: Rational) {
+                assert_eq!(a + b, c);
+            }
+            // fn test_u8(a: u8, b: Rational, c: Rational) {
+            //     assert_eq!(a + b, c);
+            // }
+            // fn test_i16(a: i16, b: Rational, c: Rational) {
+            //     assert_eq!(a + b, c);
+            // }
+
+            test_isize(-2, _1, _NEG1);
+            test_i8(-2, _1, _NEG1);
```

Comment:
Since you've only tested with integers here and elsewhere, the math problems I mentioned with `Add` and `Sub` didn't come into question.  (The denominators here are just 1.)